src/Spectre.Database/Utils/PathFinder.cs

```diff
@@ -0,0 +1,100 @@
+﻿/*
+ * PathFinder.cs
+ * Class for translating hash and friendly name to name of the file.
+ *
+   Copyright 2017 Roman Lisak
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Spectre.Database.Entities;
+
+namespace Spectre.Database.Utils
+{
+    /// <summary>
+    /// Class for finding the path of chosen property.
+    /// </summary>
+    public class PathFinder
```

Comment:
PathFinder is some sort of Repository class. It accesses the DB to return Dataset location by name or hash. Why not extract it to an interface and make it a service in case other classes would need to use that as well.